### DISCUSSION TITLE
Why does it need to constantly redownload stuff?

### DISCUSSION THREAD

**Discussion by gerroon** [tin2tin/Pallaidium#74] [Q&A] [answered]
Hi

It seems like the addon downloading same things over and over like the xl-refiner or some other check points. Is there a way to download all the downloads at once so that I do not need an internet connection all the time, also that adds up with wait time.

thanks

**Comment by tin2tin** [comment, marked as answer]
The file handling is done automatically by the Diffusers python module. For me it is only downloading when there is a new version of the model out. Unfortunately it doesn't delete the old one, so this has to be done manually. 

So, once in a while, it is a good idea to go through the snapshot folders and delete the older ones ex.: 
`C:\Users\user_name\.cache\huggingface\hub\models--stabilityai--stable-diffusion-xl-refiner-1.0\snapshots`

Downloading everything will be a crazy big download, since there are so many models available in Palladium, so downloading when needed, is properly the best solution.
